src/test/kotlin/org/rust/ide/lineMarkers/RsMethodLineSeparatorProviderTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.lineMarkers
+
+import com.intellij.codeInsight.daemon.DaemonCodeAnalyzerSettings
+
+class RsMethodLineSeparatorProviderTest : RsLineMarkerProviderTestBase() {
+
+    fun `test show`() {
+        doTest("""
+            struct Foo {} // - Has implementations
+            impl Foo {
+                const C1: i32 = 1;
+                fn f1() {}
+                fn f2() {}
+                fn f3() { // - null
+                }
+                fn f4() { // - null
+                }
+                fn f5() {} // - null
+            }
+        """)
```

Comment:
Let's add tests for functions outside of impl block: top level function, trait methods, functions in extern block